st2client/st2client/commands/action.py

```diff
@@ -503,6 +503,26 @@ def transform_object(value):
             return result
 
         def transform_array(value):
+            # Sometimes an array parameter only has a single element:
+            #
+            #     i.e. "st2 run foopack.fooaction arrayparam=51"
+            #
+            # Normally, json.loads can handle this, but not if the single element is
+            # an int.
+            #
+            # We want to try to handle this first, so it doesn't get accidentally
+            # sent to the API as an int, instead of an array of single-element int.
+            try:
+                # Force this to be a list containing the single int, then
+                # cast the whole thing to string so json.loads can handle it
+                value = str([int(value)])
```

Comment:
That's a good idea! I opened https://github.com/StackStorm/st2/issues/3160 to track this.